### ISSUE TITLE
#733contingency question and Rule Processing

### ISSUE THREAD

**Issue by yuyewuye** [sitewhere/sitewhere#821]
Hi，I also found in the run that the data could not be persisted，maybe #733  is not fully functioning as expected。If the data cannot be persisted, the system may not function properly。Please fix it ，because it‘s very importance。
And another question ， Rule Processing i don‘t konw how to use it，and the Microservice is working properly?If not could you give me a sample to how to integration Siddhi?
Thank you very much ,please reply.

**Comment by derekadams** [comment]
Hi @yuyewuye. The rule processing microservice is not fully implemented in SiteWhere 2.x, but was left in as a placeholder so that we can get back to implementing it. We will be adding it to SiteWhere 3.x before GA.

**Comment by yuyewuye** [comment]
use siddhi ?

**Comment by derekadams** [comment]
Yes, we have existing code from earlier releases that uses Siddhi, but their APIs have been updated, so we need to update our code to use them. We also are supporting [Nifi](https://nifi.apache.org/) in SiteWhere 3.0, which will allow for some other rule processing options.
